### ISSUE TITLE
Skiff Mail unable to route messages properly

### ISSUE THREAD

**Issue by x80486** [skiff-org/skiff-apps#112]
## Describe the Bug

This is going to be an interesting one :smirk:

I have `DuckDuckGo Email Protection`. It forwards e-mails to one of my `Skiff Mail` aliases. Any time I receive an e-mail from `Craiglist` (which they also hide the real e-mail) I'm unable to answer. The reply gets lost and somehow it's never received. This puzzles me because I was expecting at least an error from `Craiglist` or `Skiff Mail`.

I've tested this with `Skiff Mail`, `Fastmail`, and `Proton Mail` — when used as forward e-mail for `DuckDuckGo Email Protection`. This works fine in all the others, so I'm sure it's a problem somehow with `Skiff Mail`.

## To Reproduce

1. Sign up for `DuckDuckGo Email Protection` and use `Skiff Mail` as forwarding address
2. Sign up for `Craiglist` and use one of the `DuckDuckGo Email Protection` addresses
3. Create an ad in `Craiglist` — anything
4. Send an e-mail from a system other than `Skiff Mail` (or probably from a completely different account if using `Skiff`) to the newly random created `Craiglist` e-mail address

![image](https://github.com/skiff-org/skiff-apps/assets/34633617/d5a6e02e-ebdb-44f7-8516-abec1388f120)

5. Try to answer from `Skiff Mail` :grimacing:

## Expected Behavior

Recipient should get the replied e-mail.

## Additional Context

I'm using the Web interface — not desktop or mobile.

**Comment by amilich** [comment]
Does it go to spam inside Skiff?

**Comment by x80486** [comment]
No, I receive the e-mail on `Skiff` correctly — right in the `Inbox`. Problem is only when I replying. It sends the e-mail, but nothing is received. Not sure how is that even possible.

This may be unrelated, but I've noticed that whenever I reply, `Skiff` does not recognizes correctly (or at least it's not reflected in the UI) the e-mail alias where the message was sent to; it uses the account/main address instead.

Regardless, using any of the `Skiff` aliases or the main account e-mail yields the same result for this case.
